mitmproxy/io/io.py

```diff
@@ -23,27 +26,41 @@ def add(self, f: flow.Flow) -> None:
 
 class FlowReader:
     def __init__(self, fo: BinaryIO):
-        self.fo: BinaryIO = fo
+        self.fo: BufferedReader = BufferedReader(fo)
 
     def stream(self) -> Iterable[flow.Flow]:
         """
         Yields Flow objects from the dump.
         """
-        try:
-            while True:
-                # FIXME: This cast hides a lack of dynamic type checking
-                loaded = cast(
-                    dict[Union[bytes, str], Any],
-                    tnetstring.load(self.fo),
+
+        if self.fo.peek(1).startswith(b"{"):
+            try:
+                har_file = json.loads(self.fo.read().decode("utf-8"))
+            except Exception:
+                raise exceptions.FlowReadException(
+                    "Unable to read HAR file. Please provide a valid HAR file"
                 )
-                try:
-                    yield flow.Flow.from_state(compat.migrate_flow(loaded))
-                except ValueError as e:
-                    raise exceptions.FlowReadException(e) from e
-        except (ValueError, TypeError, IndexError) as e:
-            if str(e) == "not a tnetstring: empty file":
-                return  # Error is due to EOF
-            raise exceptions.FlowReadException("Invalid data format.") from e
+
+            for request_json in har_file["log"]["entries"]:
+                har_flow = ReadHar().request_to_flow(request_json)
```

Comment:
What about errors raised here? They should probably also be reraised as a FlowReadException?